lib/neovim-plugin.nix

```diff
@@ -157,7 +164,37 @@
                 ])
                 ++ (lib.optionals hasConfigAttrs [
                   (lib.optionalAttrs callSetup { ${namespace}.${name}.luaConfig.content = setupCode; })
-                  (lib.optionalAttrs (configLocation != null) (setLuaConfig cfg.luaConfig.content))
+                  (lib.mkIf (!cfg.lazyLoad.enable) (
+                    lib.optionalAttrs (configLocation != null) (setLuaConfig cfg.luaConfig.content)
+                  ))
+                ])
+                ++ (lib.optionals hasConfigAttrs [
+                  (lib.mkIf (cfg.lazyLoad.backend == "lz.n") {
+                    plugins.lz-n = {
+                      # infinite recursion?
+                      # enable = true;
+                      plugins = [
+                        {
+                          # TODO: handle this for every plugin properly
+                          __unkeyed-1 = originalName;
```

Comment:
I think I see why https://github.com/nix-community/nixvim/pull/1875/files#diff-639400266149b8173dbdc1cf49e1622129bfdc587b52ac1547fc4544e96e415aR46-R49 was implemented, that way... when I was running this code without checking if a lazyLoad.backend was defined, I saw errors due to rtp path depending on which attribute was used.

Reply:
Isn't this all we need?
```suggestion
                          __unkeyed-1 = luaName;
```
I don't think we need a separate lazy-specific name, we just need to use the same name as would be used in `require()` calls. Right?

Reply:
I'll try it again, last time I saw failures on some plugins. Which is why I think they had an argument  in case something needed an override.

Reply:
I think the main issue is it doesn't use the lua module name, its using the pack path directory name. I just tested on my own config and most plugins fail to be detected with `luaName` because we add the derivation to the packDir with the source name. 
![image](https://github.com/user-attachments/assets/ee84a562-80e3-4433-9ef7-6415fead62b9)

Reply:
> I think the main issue is it doesn't use the lua module name, its using the pack path directory name.

Interesting - I must've misunderstood how the rtp & lua modules work because I assumed the lua module name was derived from the directory name!

Reply:
It basically looks like rtp directory is the `originalName` and the `luaName` is the actual lua module from that directory.

Reply:
Making me think they should get better names on our end then... "dirName" and "moduleName" perhaps?